dash/include/dash/internal/TypeInfo.h

```diff
@@ -16,6 +16,13 @@ std::string typestr(const T & obj) {
          );
 }
 
+template <class T>
+std::string typestr() {
+  return dash::internal::demangle(
+           typeid(T).name()
+         );
+}
+
```

Comment:
See my comment above. I moved `dash::internal::typestr` to `dash::typestr` in header `dash/meta/TypeInfo.h`, it's documented and part of the public API now.